pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -109,35 +84,147 @@ class BudgetOptimizer(BaseModel):
         arbitrary_types_allowed=True,
     )
 
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
This is not being used anywhere. Is it needed?

Reply:
This is to allow non-native Python types (like `mmm_model: MMM`)

Reply:
I mean, do we need `model_config` in the BudgetOptimizer

Reply:
This is needed as we are using a Pydantic `BaseModel` to do type-checking